src/compilation/build_module.cc

```diff
@@ -0,0 +1,317 @@
+/*!
+ *  Copyright (c) 2017 by Contributors
+ *  Compile executable modules.
+ * \file build_module.cc
+ */
+#include "./tvm/build_module.h"
+#include "./tvm/operation.h"
+#include "./tvm/ir_pass.h"
+#include "./tvm/codegen.h"
+
+
+namespace tvm {
+
+std::string Target::str() const {
+    std::ostringstream result;
+    result << target_name;
+    for (const auto &x : options) {
+        result << " " << x;
+    }
+    return result.str();
+}
+
+
+Target TargetFromName(const std::string& name) {
+    if (name == "llvm") {
+        return target::llvm();
+    } else if (name == "cuda" || name == "nvptx") {
+        return target::cuda();
+    } else if (name == "rocm" || name == "opencl") {
+        /* For now, assume rocm schedule for opencl */
+        return target::rocm();
+    } else if (name == "metal") {
+        return target::metal();
+    } else if (name == "stackvm" || name == "ext_dev") {
+        return target::stackvm();
+    } else {
+        LOG(ERROR) << "Unknown target name " << name;
+        return target::stackvm();
+    }
+}
+
+bool StartsWith(const std::string& str, const std::string& pattern) {
+    return str.compare(0, pattern.length(), pattern) == 0;
+}
+
+std::string GetDeviceName(const std::string& target_str) {
+    std::stringstream ss(target_str);
+    std::string item;
+    /* skip target name */
+    std::getline(ss, item, ' ');
```

Comment:
you can simply use ss >> item.  for option parsing, refer also refer to https://github.com/dmlc/tvm/blob/master/src/codegen/llvm/llvm_common.cc#L53